### ISSUE TITLE
OptionSettings fail to be applied

### ISSUE THREAD

**Issue by sidewinder12s** [hsingh/ansible-elastic-beanstalk#21]
Using option settings appears to fail with the new elasticbeanstalk_env module:

TASK [Deploy Application Version to Elastic Beanstalk Environment] *************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Invalid type for parameter OptionSettings[42], value: ('aws:elasticbeanstalk:container:tomcat:jvmoptions', 'XX:MaxPermSize', '256m'), type: <type 'tuple'>, valid types: <type 'dict'>

I believe it is this line: https://github.com/hsingh/ansible-elastic-beanstalk/blob/master/library/elasticbeanstalk_env.py#L324-L325

Boto3 doesn't require tuples anymore so you can just directly pass the option_settings dict to the API call without doing any kind of conversion: http://boto3.readthedocs.io/en/latest/reference/services/elasticbeanstalk.html#ElasticBeanstalk.Client.create_environment

I can open a PR if you want 😄

**Comment by nergdron** [comment]
I'm seeing something similar that's probably the same problem:

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: string indices must be integers, not str
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_6A6cMp/ansible_module_elasticbeanstalk_env.py\", line 414, in <module>\n    main()\n  File \"/tmp/ansible_6A6cMp/ansible_module_elasticbeanstalk_env.py\", line 324, in main\n    option_setting_tups = [(os['Namespace'],os['OptionName'],os['Value']) for os in option_settings]\nTypeError: string indices must be integers, not str\n", "module_stdout": "", "msg": "MODULE FAILURE"}
```

**Comment by sidewinder12s** [comment]
Your task option_settings dict's include an integer that it is blowing up
on? :)

**Comment by hsingh** [comment]
Will address it this weekend.

**Comment by nergdron** [comment]
I'm seeing a related issue, where I'm passing in a list of dicts which looks like this:

```
ok: [localhost] => {
    "eb_all_options": [
        {
            "Namespace": "aws:elb:policies", 
            "OptionName": "ConnectionDrainingEnabled", 
            "Value": "True"
        }, 
        {
            "Namespace": "aws:elb:policies", 
            "OptionName": "ConnectionDrainingTimeout", 
            "Value": "300"
        }, 
        {
            "Namespace": "aws:elb:policies", 
            "OptionName": "ConnectionSettingIdleTimeout", 
            "Value": "600"
        }, 
  ]
}
```
but the module blows up when passing it in:
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: Invalid type for parameter OptionSettings[2], value: ('aws:elb:policies', 'ConnectionSettingIdleTimeout', '600'), type: <type 'tuple'>, valid types: <type 'dict'>
```

Seems like it's converting things to tuples when the API wants a dict anyway?

So doing a straight passthrough of the options list to boto would likely fix this for me as well.

**Comment by hsingh** [comment]
Fixed by #22

**Comment by nergdron** [comment]
Such quick turnaround on this fix, things are working great for me now. Thanks so much! :)
